src/migrations/20230807000000-genericReport-migrateTrainingReports.js

```diff
@@ -0,0 +1,463 @@
+        UNION ALL
+        SELECT
+          srp.id pilot_record_id,
+          2 "reportTypeId", -- the 'report.trainingSession' ID in ValidFor
+          s.id "statusId",
+          srp.data->>'context' context,
+          to_date(srp.data->>'startDate','MM/DD/YYYY') "startDate",
+          to_date(srp.data->>'endDate','MM/DD/YYYY') "endDate",
+          srp."createdAt",
+          srp."updatedAt",
+          srp.data
+        FROM "SessionReportPilots" srp
+        JOIN "Statuses" s
+          ON srp.data->>'status' = s.name
+          AND s."validForId" = 2
+        ;
+
+        -- Create Report records in a janky way to get the new and old IDs mapped
+        INSERT INTO "Reports" (
+          "reportTypeId",
+          "statusId",
+          context,
+          "startDate",
+          "endDate",
+          "createdAt",
+          "updatedAt"
+        )
+        SELECT
+          "reportTypeId",
+          "statusId",
+          COALESCE(context, 'null') || 'XYX' || pilot_record_id,
+          "startDate",
+          "endDate",
+          "createdAt",
+          "updatedAt"
+        FROM interim_reports
+        ;
+
+        ALTER TABLE interim_reports ADD COLUMN reports_id INTEGER;
+
+        -- add the Reports.id mapping to interim reports so we can use it elsewhere
+        UPDATE interim_reports
+        SET reports_id = id
+        FROM "Reports" r
+        WHERE (string_to_array(r.context, 'XYX'))[2]::integer = pilot_record_id
+        ;
+
+        -- Set Reports.context back to its normal values
+        UPDATE "Reports"
+        SET context = NULLIF((string_to_array(context, 'XYX'))[1], 'null'
+        ;
+
+      
+        INSERT INTO "ReportTrainingEvents"
+        (
+          "reportId",
+          "regionId",
+          "eventId",
+          name,
+          "organizerId",
+          audience,
+          "trainingType",
+          vision,
+          "createdAt",
+          "updatedAt"
+        )
+        SELECT
+        FROM "EventReportPilots" erp
+
+        INSERT INTO "ReportTrainingSessions"(
+          "reportId",
+          "reportTrainingEventId",
+          "regionId",
+          name,
+          "createdAt",
+          "updatedAt"
+        )
+        SELECT
+          
+        FROM "SessionReportPilots" srp
+        JOIN interim_reports ir
+          ON pilot_record_id = srp.id
+          AND ir."reportTypeId" = 2
+        JOIN "Reports" r
+          ON srp.id = pilot_record_id
+
+        --
+        INSERT INTO "ReportTargetPopulations" (
+          "reportId",
+          "targetPopulationId",
+          "createdAt",
+          "updatedAt"
+        )
+        WITH unnested_target_populations AS (
+        SELECT
+          id,
+          UNNEST(
+            string_to_array(jsonb_array_elements_text(data->'targetPopulations'),E'\n')
+          ) target_population
+        FROM "EventReportPilots" erp
+        )
+        SELECT
+          reports_id,
+          utp.id,
+          "createdAt",
+          "updatedAt"
```

Comment:
@hardwarehuman: I know is not explicitly needed, but please include the table alias prefix to the selected columns so its easier to follow were each field comes from.